make_fx can now SymIntify int inputs

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113921
* __->__ #113452
* #113911
* #113877

This PR also contains a basket of fixes that were turned up by now testing more arguments with SymInt. I fixed as many of the easy ones as I could easily get earlier in this stack and a bunch here, but there are some more annoying ones I xfailed.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

ezyang committed:
make_fx can now SymIntify int inputs

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

ezyang committed:
Update on "make_fx can now SymIntify int inputs"


Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

Chillee reviewed (approved):
test?

ezyang committed:
Update on "make_fx can now SymIntify int inputs"


Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang commented:
added test

ezyang committed:
Update on "make_fx can now SymIntify int inputs"


Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang committed:
Update on "make_fx can now SymIntify int inputs"


Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang commented on test/test_proxy_tensor.py:
```diff
@@ -1705,11 +1715,15 @@ def f(t):
+    xfail('linalg.norm', ''),  # AssertionError: False != True - https://github.com/pytorch/pytorch/issues/113905
+    xfail('linalg.norm', 'subgradients_at_zero'),  # https://github.com/pytorch/pytorch/issues/113905
+    xfail('max_pool2d_with_indices_backward', ''),  # Expected a value of type 'List[int]' for argument 'kernel_size' but...
+    xfail('randint_like', ''),  # when unpacking SymInt, expected int but got s0
```
this one is easy but involves a C++ header change, going to be next

ezyang commented on test/test_proxy_tensor.py:
```diff
@@ -1705,11 +1715,15 @@ def f(t):
 
+    xfail('linalg.norm', ''),  # AssertionError: False != True - https://github.com/pytorch/pytorch/issues/113905
+    xfail('linalg.norm', 'subgradients_at_zero'),  # https://github.com/pytorch/pytorch/issues/113905
+    xfail('max_pool2d_with_indices_backward', ''),  # Expected a value of type 'List[int]' for argument 'kernel_size' but...
```
This one involves a fairly involved refactor of the JIT IValue parsing code

Chillee commented:
You're getting failures since the testing now involves significantly more symints right?

ezyang commented:
Yeah, we are now actually feeding symints for all the int arguments, as opposed to only Tensor shapes now

ezyang committed:
Update on "make_fx can now SymIntify int inputs"


This PR also contains a basket of fixes that were turned up by now testing more arguments with SymInt. I fixed as many of the easy ones as I could easily get earlier in this stack and a bunch here, but there are some more annoying ones I xfailed.

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang committed:
Update on "make_fx can now SymIntify int inputs"


This PR also contains a basket of fixes that were turned up by now testing more arguments with SymInt. I fixed as many of the easy ones as I could easily get earlier in this stack and a bunch here, but there are some more annoying ones I xfailed.

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge
